File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/media/files/cached/configuration/IDiskFileCacheConfiguration.java
package com.lasthopesoftware.bluewater.client.library.items.media.files.cached.configuration;

import com.lasthopesoftware.bluewater.client.library.repository.Library;

import org.joda.time.Duration;

public interface IDiskFileCacheConfiguration {
	String getCacheName();
	Library getLibrary();
	long getMaxSize();
	Duration getCacheItemLifetime();
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/items/media/files/cached/configuration/IDiskFileCacheConfiguration.kt
package com.lasthopesoftware.bluewater.client.browsing.items.media.files.cached.configuration

import com.lasthopesoftware.bluewater.client.browsing.library.repository.Library
import org.joda.time.Duration

interface IDiskFileCacheConfiguration {
    val cacheName: String
    val library: Library
    val maxSize: Long
    val cacheItemLifetime: Duration?
}
